### PR TITLE
Delete legacy docs and add new project for reStructuredText

### PR DESCRIPTION
**Pull request by neewy**
### Description of the Change

The purpose of this documentation is to convey design and architecure aspects of Iroha ledger in a sctructured approach, as well as operational side: how-tos, guides, and examples. Docs are accessible via ReadTheDocs website, and can be generated to a lot of formats, available in Sphinx. In order to contribute, one should to be familiar with [reStructuredTest](docutils.sourceforge.net/rst.html) syntax, and follow principles described in this file.

### Benefits

Deleted legacy, created a structure for docs and example of docs, including launching Iroha on single node.

### Possible Drawbacks 

Requires constant updates of docs, and administering project in ReadTheDocs.

### Usage Examples or Tests *[optional]*

 * Python 2.7 installed
 * `pip install -r requirements.txt` in docs/source

Build steps

1. `cd docs/source`
1. `make html`

After that, the documentation is generated in html format in `_build` folder. You can check another formats on Sphinx website or just by typing `make`.

### CODE REVIEW

**Comment by Warchant** [comment]
Reviewed 38 of 39 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 11 unresolved discussions, some commit checks broke.

---

*[docs/source/.gitignore, line 2 at r2](https://reviewable.io:443/reviews/hyperledger/iroha/884#-L3E1Je-kA8rtdBPEPra:-L3E1Je-kA8rtdBPEPrb:b-202xa1) ([raw file](https://github.com/hyperledger/iroha/blob/19bf3c339123bef66c391d3a26d7422737e5d21e/docs/source/.gitignore#L2)):*
> ```
> _build
> .vscode
> ```

What about sphinx generated files?

---

*[docs/source/conf.py, line 170 at r2](https://reviewable.io:443/reviews/hyperledger/iroha/884#-L3E1RzYlOfQLbkZv1gZ:-L3E1RzYlOfQLbkZv1g_:b-1jiay4) ([raw file](https://github.com/hyperledger/iroha/blob/19bf3c339123bef66c391d3a26d7422737e5d21e/docs/source/conf.py#L170)):*
> ```Python
> #  dir menu entry, description, category)
> texinfo_documents = [
>     (master_doc, 'Iroha', u'Iroha Documentation',
> ```

Move "Iroha Documentation", "Iroha", 'Nikolay Yushkevich at Soramitsu Co LLC' to separate variables and reuse variables.

---

*[docs/source/deployment.rst, line 29 at r2](https://reviewable.io:443/reviews/hyperledger/iroha/884#-L3E28sIyXgly9NclaVD:-L3E28sIyXgly9NclaVE:ban0zv5) ([raw file](https://github.com/hyperledger/iroha/blob/19bf3c339123bef66c391d3a26d7422737e5d21e/docs/source/deployment.rst#L29)):*
> ```reStructuredText
> """""""""""""""""""""""""
> 
> There is a list of assumptions which you should review before proceeding:
> ```

Maybe hot *review*, but 

"There is a list of assumptions, which must hold before proceeding", or something like that?

---

*[docs/source/deployment.rst, line 31 at r2](https://reviewable.io:443/reviews/hyperledger/iroha/884#-L3E2HXUMPjN6v0BBrvJ:-L3E2HXUMPjN6v0BBrvK:bo0bfuy) ([raw file](https://github.com/hyperledger/iroha/blob/19bf3c339123bef66c391d3a26d7422737e5d21e/docs/source/deployment.rst#L31)):*
> ```reStructuredText
> There is a list of assumptions which you should review before proceeding:
> 
>  * Postgres server is up and running, the port is 5432
> ```

Port does not matter, we specify it in config

---

*[docs/source/deployment.rst, line 32 at r2](https://reviewable.io:443/reviews/hyperledger/iroha/884#-L3E2KqGF8SrkFzWZ9nN:-L3E2KqGF8SrkFzWZ9nO:buo9nmt) ([raw file](https://github.com/hyperledger/iroha/blob/19bf3c339123bef66c391d3a26d7422737e5d21e/docs/source/deployment.rst#L32)):*
> ```reStructuredText
> 
>  * Postgres server is up and running, the port is 5432
>  * Redis server is up and running, the port is 6379
> ```

Same here

---

*[docs/source/deployment.rst, line 33 at r2](https://reviewable.io:443/reviews/hyperledger/iroha/884#-L3E2MOcI4FCfKrgfKAx:-L3E2MOcI4FCfKrgfKAy:b-pxegae) ([raw file](https://github.com/hyperledger/iroha/blob/19bf3c339123bef66c391d3a26d7422737e5d21e/docs/source/deployment.rst#L33)):*
> ```reStructuredText
>  * Postgres server is up and running, the port is 5432
>  * Redis server is up and running, the port is 6379
>  * Iroha daemon binary is built and accessible in your system
> ```

better write "`irohad`  iroha daemon..."

---

*[docs/source/deployment.rst, line 34 at r2](https://reviewable.io:443/reviews/hyperledger/iroha/884#-L3E2Sba_PUzRUXGF6w5:-L3E2Sba_PUzRUXGF6w6:b-8y80cu) ([raw file](https://github.com/hyperledger/iroha/blob/19bf3c339123bef66c391d3a26d7422737e5d21e/docs/source/deployment.rst#L34)):*
> ```reStructuredText
>  * Redis server is up and running, the port is 6379
>  * Iroha daemon binary is built and accessible in your system
>  * The genesis block and configuration files were created
> ```

*are created* ?

---

*[docs/source/deployment.rst, line 37 at r2](https://reviewable.io:443/reviews/hyperledger/iroha/884#-L3E2Wl5crNsvoVqNfA9:-L3E2Wl5crNsvoVqNfAA:b-p43ija) ([raw file](https://github.com/hyperledger/iroha/blob/19bf3c339123bef66c391d3a26d7422737e5d21e/docs/source/deployment.rst#L37)):*
> ```reStructuredText
>  * Config file uses valid postgres and redis connection settings
>  * A keypair for the peer is generated
>  * This is the first time you run the Iroha on this peer and you want to create new chain
> ```

What if user wants to start not for the first time? 
Add TODO to add link to guide later.

---

*[docs/source/deployment.rst, line 124 at r2](https://reviewable.io:443/reviews/hyperledger/iroha/884#-L3E2o2NnWBBL0YpuLWY:-L3E2o2NnWBBL0YpuLWZ:b5ylx7x) ([raw file](https://github.com/hyperledger/iroha/blob/19bf3c339123bef66c391d3a26d7422737e5d21e/docs/source/deployment.rst#L124)):*
> ```reStructuredText
> """"""""""""""""""""""""""""""""""""""""
> 
> There is a list of assumptions which you should review before proceeding:
> ```

again *review*

---

*[docs/source/deployment.rst, line 125 at r2](https://reviewable.io:443/reviews/hyperledger/iroha/884#-L3E2rHL5bBWtZ_oasSa:-L3E2rHL5bBWtZ_oasSb:bfzx008) ([raw file](https://github.com/hyperledger/iroha/blob/19bf3c339123bef66c391d3a26d7422737e5d21e/docs/source/deployment.rst#L125)):*
> ```reStructuredText
> 
> There is a list of assumptions which you should review before proceeding:
>  * Postgres and redis servers are running on the same docker network
> ```

postgres and redis servers are accessible from iroha peer.
?

---

*[docs/source/install.rst, line 14 at r2](https://reviewable.io:443/reviews/hyperledger/iroha/884#-L3E3Db-R-fxys-gpJEj:-L3E3Db-R-fxys-gpJEk:b-zfw57l) ([raw file](https://github.com/hyperledger/iroha/blob/19bf3c339123bef66c391d3a26d7422737e5d21e/docs/source/install.rst#L14)):*
> ```reStructuredText
> ^^^^^^^^^^^^^^^^^^^^^^^
> 
> In order to contribute to the codebase of Hyperledger Iroha, one should meet following requirements:
> ```

Maybe add link to CONTRIBUTING.md?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hyperledger/iroha/884)*

**Comment by neewy** [comment]
> What about sphinx generated files?

There aren't any such files

> What if user wants to start not for the first time?

That is said in the attention section, which says "If you have stopped the daemon and want to use existing chain — you should not pass the genesis block parameter." Hopefully, we will change the flags and this won't be necessary to address as "Attention" note 

> Maybe add link to CONTRIBUTING.md?
no. this section will explain what kind of development environment is needed in future

> again review
It should be something like review before reading further, in order to understand the context

File: docs/source/deployment.rst [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+
+Docker
+------
+
+In order to run Iroha peer as a single instance in Docker, you should pull the image for Iroha first:
+
+.. code-block:: shell
+
+    docker pull hyperledger/Iroha-docker:latest
```
Comment (lebdron) [inline docs/source/deployment.rst]: I get `invalid reference format: repository name must be lowercase` when I run this command.

Iroha -> iroha

File: docs/source/deployment.rst [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+Automated
+---------
+
+Anyone can reuse existing Ansible Playbook in order to create a network of peers running Iroha. Currently, this is a solution for development and testing, in other words, a proof of concept, and cannot be used in production environment, due to some security flaws. For production network, a manual composing of genesis block is required.
+
+Prerequisites
+"""""""""""""
+
+ * One ore more <virtual> machines with a linux distributive installed.
```
Comment (lebdron) [inline docs/source/deployment.rst]: linux -> Linux

File: docs/source/deployment.rst [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+8. Run playbook, providing your private key and hosts file
+ 
+.. code-block:: shell
+
+    ansible-playbook --private-key=~/.ssh/iroha -i hosts provisioning.yml
+
+9. Wait until playbook finishes and then Iroha network is ready and up.
+
+Cheking iroha status
```
Comment (lebdron) [inline docs/source/deployment.rst]: Cheking -> Checking

File: docs/source/deployment.rst [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+Not having genesis block
+------------------------
+
+Create genesis block by generating it via `iroha-cli` or manually, as it is described `here <https://hyperledger.github.io/iroha-api/#create-genesis-block>`__
+
+Not having a keypair for a peer
+-------------------------------
+
+In order to create a keypair for an account or a peer, use iroha-cli binary by passing the name of the peer with -new_account option.
```
Comment (lebdron) [inline docs/source/deployment.rst]: `--new_account` should probably be inside code block.

File: docs/source/install.rst [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+Install Prerequisites
+---------------------
+
+.. Caution:: This section is in development and should be expanded later
+
+Development environment
+^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to contribute to the codebase of Hyperledger Iroha, one should meet following requirements:
```
Comment (lebdron) [inline docs/source/install.rst]: meet following -> meet the following

**Comment by Warchant** [comment]
:lgtm:. Just add line terminators, fix @lebdron's issues and it can be merged.

---

Reviewed 6 of 6 files at r4.
Review status: all files reviewed at latest revision, 23 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hyperledger/iroha/884#-:-L3YLoNm-fDiPZWGKgCy:b-9lesc6)*

**Pull request merged**